pkg/webreg/webreg.go

```diff
@@ -729,6 +729,78 @@ to configure that test to run on a schedule, instead of as a pre-submit:
+contains a test workflow, the following <code>ImageStreams</code> will exist:
+</p>
+
+<table>
+  <tr>
+    <th style="white-space: nowrap"><code>ImageStream</code></th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td style="white-space: nowrap"><code>pipeline<code></td>
+    <td>Input images described with <code>base_images</code> and <code>build_root</code> as well as images holding built artifacts and output images as defined in <code>images</code>.</td>
+  </tr>
+  <tr>
+    <td style="white-space: nowrap"><code>release<code></td>
+    <td>Tags of this <code>ImageStreams</code> hold OpenShift release payload images for installing and upgrading ephemeral OpenShift clusters for testing; a tag will be present for every named release configured in <code>releases<code>. If a <code>tag_specification<code> is provided, two tags will be present, <code>:initial</code> and <code>:latest</code>.</td>
+  </tr>
+  <tr>
+    <td style="white-space: nowrap"><code>stable-&lt;name&gt;<code></td>
+    <td>Images composing the <code>release:name</code> release payload, present when <code>&lt;name&gt;<code> is configured in <code>releases<code>.</td>
+  </tr>
+  <tr>
+    <td style="white-space: nowrap"><code>stable<code></td>
+    <td>Same as above, but for the <code>release:latest</code> release payload. Appropriate tags are overridden using the container images built during the test.</td>
+  </tr>
+</table>
+
+<h4 id="config-references"><a href="#config-references">Referring to Images in <code>ci-operator</code> Configuration</a></h4>
+<p>
+Inside of any <code>ci-operator</code> configuration file all images must be 
+referenced as an <code>ImageStreamTag</code> (<code>stream:tag</code>), but
+may be referenced simply with the tag name. When an image is referenced with
+a tag name, the tag will be resolved on the <code>pipeline</code> <code>ImageStream</code>,
+if possible, falling back to the <code>stable</code> <code>ImageStream</code>
+if not. For example, an image referenced as <code>installer</code> will use
+<code>pipeline:installer</code> if that tag is present, falling back to
+<code>stable:installer</code> if not. The following configuration fields
+use this defaulting mechanism:
+</p>
+
+<ul>
+  <li><code>images[*].from</code>: configuring the base <code>FROM</code> which an image builds</li>
+  <li><code>promotion.additional_images</code>: configuring which images are published</li>
+  <li><code>promotion.excluded_images</code>: configuring which images are not published</li>
+  <li><code>tests[*].container.from</code>: configuring the container image in which a single-stage test runs</li>
+  <li><code>tests[*].steps.{pre,test,post}[*].from</code>: configuring the container image which some part of a multi-stage test runs</li>
+</ul>
+
+<h4 id="literal-references"><a href="#literal-references">Referring to Images in Tests</a></h4>
+<p>
+In some cases, it is necessary for a test command to refer to an image that
```

Comment:
I am confused as to how this works. My understanding so far has been that there is some ~magic~ logic that makes sure the four imagestreams described above exist prior to running e2e tests and that this is only needed if one step needs an image from its own config that it doesn't explicitly rely on. Could you maybe give an example?